### PR TITLE
Docs: Add text-to-uppercase variable syntax info 

### PR DESCRIPTION
**Pull request by hoobtron**
new Text to uppercase FormatRegistryItem

**What is this feature?**

Variable Text to uppercase

Formats single- and multi-valued variables into their uppercase text representation. For a single variable it will just return the uppercase text representation. For multi-valued variables it will return the uppercase text representation combined with `+`.

This PR is a in combination with pull Scenes pull requests [746](https://github.com/grafana/scenes/pull/746) and [747](https://github.com/grafana/scenes/pull/747).

**Why do we need this feature?**

https://github.com/grafana/scenes/issues/441

**Who is this feature for?**

Dashboard/panel editors

**Which issue(s) does this PR fix?**:

Fixes #441

**Special notes for your reviewer:**

Please check that:
- [ ] include/approve https://github.com/grafana/scenes/pull/746 & https://github.com/grafana/scenes/pull/747
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/dashboards/variables/variable-syntax/index.md [hunk 1]
```diff
@@ -184,6 +184,16 @@ String to interpolate: '${servers:text}'
 Interpolation result: "test1 + test2"
 ```
 
+### Text to uppercase
+
+Formats single- and multi-valued variables into their uppercase text representation. For a single variable it will just return the uppercase text representation. For multi-valued variables it will return the uppercase text representation combined with `+`.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/variable-syntax/index.md:189]: I know this isn't like the other entries, but I'm updating the wording to follow our style guide. The page as a whole will eventually be edited for that as well.
```suggestion
Formats single- and multi-valued variables into their uppercase text representation. For a single variable it just returns the uppercase text representation. For multi-valued variables it returns the uppercase text representation combined with `+`.
```

**Review by imatwawana — COMMENTED** [review]
Thanks for the contribution! In terms of the wording I've suggested a couple tweaks. I'll tag the Dashboards team for technical review.

**Review by axelavargas — COMMENTED** [review]
Hey @hoobtron 👋🏾 , thanks for contributing to Grafana 🏅 . I see there are some suggestions left by @imatwawana , when you have time, please review and commit them :) 

Also, in the scenes PR: https://github.com/grafana/scenes/pull/746 , we left some comments, we will need to merge that PR before merging this one

P.S:  We are currently cleaning our PR review queue, it might happen that in ~2 weeks this PR gets closed, however, if you made the changes the PR can be reopened :) 

Thank you 🙏🏾

**Comment by hoobtron** [comment]
Hi, I accepted text change suggestion and merged PR 747 into 746.

Is there something else to do for me?

**Comment by axelavargas** [comment]
> Hi, I accepted text change suggestion and merged PR 747 into 746.
> 
> Is there something else to do for me?

Nothing else, we will need to merge the other PR first :)

**Comment by nils-** [comment]
I've run into an issue that I need to have upper/lowercase variables and this would be nice to have.

**Pull request closed without merging**